lib/tools/system-calls.js

```diff
@@ -554,16 +554,30 @@ systemCallMethods.reboot = async function (retries = DEFAULT_ADB_REBOOT_RETRIES)
       }
     });
   } finally {
-    if (adbdRooted) {
-      try {
-        await exec(this.executable.path, ['unroot']);
-      } catch (err) {
-        log.warn(`Unable to unroot adb daemon: '${err.message}'. Continuing`);
-      }
+    if (this.adbdRooted) {
+      this.unroot();
     }
   }
 };
 
+systemCallMethods.root = async function () {
+  try {
+    await exec(this.executable.path, ['root']);
+    this.adbdRooted = true;
```

Comment:
I think there is no point to maintain **this.adbdRooted** flag in these method if we ignore errors. I's rather leave it there where it was
Also, it might be useful to return true/false from these methods to be able to understand whether the operation was successful or not from the client code